databroker/src/grpc/kuksa_val_v2/val.rs

```diff
@@ -460,4 +539,130 @@ mod tests {
             }
         }
     }
+
+    #[tokio::test]
+    async fn test_list_metadata_using_wildcard() {
+        let broker = DataBroker::default();
+        let authorized_access = broker.authorized_access(&permissions::ALLOW_ALL);
+
+        authorized_access
+            .add_entry(
+                "test.datapoint1".to_owned(),
+                broker::DataType::Bool,
+                broker::ChangeType::OnChange,
+                broker::EntryType::Sensor,
+                "Test datapoint 1".to_owned(),
+                None,
+                None,
+            )
+            .await
+            .expect("Register datapoint should succeed");
+
+        authorized_access
+            .add_entry(
+                "test.branch.datapoint2".to_owned(),
+                broker::DataType::Bool,
+                broker::ChangeType::OnChange,
+                broker::EntryType::Sensor,
+                "Test branch datapoint 2".to_owned(),
+                None,
+                None,
+            )
+            .await
+            .expect("Register datapoint should succeed");
+
+        let mut wildcard_req = tonic::Request::new(proto::ListMetadataRequest {
+            root: "test.**".to_owned(),
+            filter: "".to_owned(),
+        });
+
+        // Manually insert permissions
+        wildcard_req
+            .extensions_mut()
+            .insert(permissions::ALLOW_ALL.clone());
+
+        match proto::val_server::Val::list_metadata(&broker, wildcard_req)
+            .await
+            .map(|res| res.into_inner())
+        {
+            Ok(list_response) => {
+                let entries_size = list_response.metadata.len();
+                assert_eq!(entries_size, 2);
+            }
+            Err(_status) => panic!("failed to execute get request"),
+        }
```

Comment:
add another test with only one asteriks?

Reply:
You are right, extended unit test to support one asteriks.